augur/ancestral.py

```diff
@@ -351,27 +369,17 @@ def run(args):
     write_json(anc_seqs, out_name)
     print("ancestral mutations written to", out_name, file=sys.stdout)
 
-    if args.output_sequences:
-        if args.output_vcf:
-            # TODO: This should be an error and we should check for this
-            # unsupported combination of arguments at the beginning of the
-            # script to avoid wasting time for users.
-            print("WARNING: augur only supports sequence output for FASTA alignments and not for VCFs.", file=sys.stderr)
-        else:
-            records = [
-                SeqRecord(Seq(node_data["sequence"]), id=node_name, description="")
-                for node_name, node_data in anc_seqs["nodes"].items()
-            ]
-            SeqIO.write(records, args.output_sequences, "fasta")
-            print("ancestral sequences FASTA written to", args.output_sequences, file=sys.stdout)
-
-    # If VCF, output VCF including new ancestral seqs
-    if is_vcf:
-        if args.output_vcf:
-            vcf_fname = args.output_vcf
-        else:
-            vcf_fname = '.'.join(args.alignment.split('.')[:-1]) + '.vcf'
-        write_vcf(nuc_result['tt'].get_tree_dict(keep_var_ambigs=True), vcf_fname)
-        print("ancestral sequences as vcf-file written to",vcf_fname, file=sys.stdout)
+    if not is_vcf and args.output_sequences:
+        records = [
+            SeqRecord(Seq(node_data["sequence"]), id=node_name, description="")
+            for node_name, node_data in anc_seqs["nodes"].items()
+        ]
+        SeqIO.write(records, args.output_sequences, "fasta")
+        print("ancestral sequences FASTA written to", args.output_sequences, file=sys.stdout)
+
+    # If VCF output VCF including new ancestral seqs if requested to
+    if is_vcf and args.output_vcf:
```

Comment:
Suggestion: keep the conditions simple and use `assert` for checking if the file is VCF, now that the checks are happening at the beginning.

```py
if args.output_sequences:
    assert not is_vcf

    records = [
        SeqRecord(Seq(node_data["sequence"]), id=node_name, description="")
        for node_name, node_data in anc_seqs["nodes"].items()
    ]
    SeqIO.write(records, args.output_sequences, "fasta")
    print("ancestral sequences FASTA written to", args.output_sequences, file=sys.stdout)

# output VCF including new ancestral seqs
if args.output_vcf:
    assert is_vcf

    write_vcf(nuc_result['tt'].get_tree_dict(keep_var_ambigs=True), args.output_vcf)
    print("Mutations, including for ancestral nodes, exported as VCF to", args.output_vcf, file=sys.stdout)
```

Reply:
Changed - thanks!